src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -166,18 +159,18 @@ public async Task<T> GetConfigurationAsync(CancellationToken cancel)
                         // Don't use the individual CT here, this is a shared operation that shouldn't be affected by an individual's cancellation.
                         // The transport should have it's own timeouts, etc..
                         var configuration = await _configRetriever.GetConfigurationAsync(MetadataAddress, _docRetriever, CancellationToken.None).ConfigureAwait(false);
-                        _lastRefresh = DateTimeOffset.UtcNow;
-                        // Add 1 hour jitter to avoid spike traffic to IdentityProvider.
-                        _syncAfter = DateTimeUtil.Add(DateTime.UtcNow, AutomaticRefreshInterval + TimeSpan.FromMinutes(new Random().Next(60)));
                         if (_configValidator != null)
                         {
                             ConfigurationValidationResult result = _configValidator.Validate(configuration);
                             if (!result.Succeeded)
-                                LogHelper.LogWarning(LogMessages.IDX20810, result.ErrorMessage);
+                                throw LogHelper.LogExceptionMessage(new InvalidConfigurationException(LogHelper.FormatInvariant(LogMessages.IDX20810, result.ErrorMessage)));
                         }
 
+                        _lastRefresh = DateTimeOffset.UtcNow;
+                        // Add up to 1 hour jitter to avoid spike traffic to IdentityProvider.
+                        _syncAfter = DateTimeUtil.Add(DateTime.UtcNow, AutomaticRefreshInterval +
+                            TimeSpan.FromSeconds(new Random().Next((int)Math.Min(AutomaticRefreshInterval.TotalSeconds / 10, TimeSpan.FromSeconds(600).TotalSeconds))));
                         _currentConfiguration = configuration;
```

Comment:
It seems to me we should use the same jitter strategy we used for `_bootstrapRefreshIntervalWithJitter` in a recent PR. Another possibility is we should update that jitter strategy to match this. In any case we shouldn't have jitter behave one way in one section and another way in another section.

Here we seem to be saying Interval + 0-Interval/10 and in the other case we're saying 0-Interval.

Further, we shoudln't use "magic numbers". Since we're picking (random?) numbers here let's at least comment on the reasoning to make revisiting easier--this may be moot if we follow the older pattern.

I'm not necessarily against having a floor (i.e. the min range vs constant value) but in that case the constant value should be something declared up top not just inline here.

Reply:
Updated to 0~5%